### PR TITLE
Small fixes to changing duration of neighbour consultation

### PR DESCRIPTION
**Pull request by cdccollins**
### Description of change

- Don't replace placeholder values in neighbour letter to avoid confusion over setting new dates for when consultation will end - use consultee pattern

### Story Link

https://trello.com/c/7gIhVzv1/2474-able-to-change-the-duration-of-neighbour-consultation-period

### CODE REVIEW

File: app/models/consultation.rb [hunk 1]
```diff
@@ -259,15 +266,22 @@ def neighbour_letter_body
...
     "# #{neighbour_letter_header}\n\n#{neighbour_letter_body}"
   end
 
   def neighbour_letter_text
-    super.presence || neighbour_letter_content
+    if super.presence&.include?("{{")
+      neighbour_letter_content
+    else
+      super.presence
+    end
```
Comment (cdccollins) [inline app/models/consultation.rb:284]: Obviously not very happy with this. Can someone suggest a better way?

File: app/views/planning_applications/neighbour_letters/_letter_template.html.erb [hunk 2]
```diff
@@ -33,7 +33,7 @@
         </p>
       <% end %>
     </div>
-    <%= form.text_area :neighbour_letter_text, rows: 25, class: "govuk-body neighbour-letter-template" %>
+    <%= form.text_area :neighbour_letter_text, value: "#{I18n.t("neighbour_letter_template.#{planning_application.application_type.name}")}", rows: 25, class: "govuk-body neighbour-letter-template" %>
```
Comment (cdccollins) [inline app/views/planning_applications/neighbour_letters/_letter_template.html.erb:36]: Also not convinced this is the best way 😬

File: app/models/consultation.rb (hunk 1 above)
Comment (benjamineskola) [reply app/models/consultation.rb:284]: I wonder if it's worth just unconditionally using `neighbour_letter_content`? Is there any downside to running the `replace_placeholders` part if there are no placeholders in it? I assume there will be _some_ performance impact, but probably tiny, relatively speaking.

File: app/views/planning_applications/neighbour_letters/_letter_template.html.erb (hunk 2 above)
Comment (benjamineskola) [reply app/views/planning_applications/neighbour_letters/_letter_template.html.erb:36]: I don't think the string interpolation is needed, if we're interpolating the entire string.

```suggestion
    <%= form.text_area :neighbour_letter_text, value: I18n.t("neighbour_letter_template.#{planning_application.application_type.name}"), rows: 25, class: "govuk-body neighbour-letter-template" %>
```

**Review by benjamineskola — APPROVED** [review]

File: app/models/consultation.rb (hunk 1 above)
Comment (cdccollins) [reply app/models/consultation.rb:284]: The problem with that is it then overwrites any edits that the officer makes in the text box

**Pull request merged**